web/docker/README.md

```diff
@@ -0,0 +1,100 @@
+# Docker
+Docker provides a way to run applications securely isolated in a container,
+packaged with all its dependencies and libraries.
+
+To see how you can run CodeChecker server in docker read the following
+sections.
+
+## Build docker image
+You can create a docker image by using the following command:
+```bash
+docker build -t codechecker-server:latest .
+```
+
+Multiple build-time variables can be specified:
+- `CC_VERSION` (default: *master*): CodeChecker git tag version or use `master`
+to build an image from the latest CodeChecker.
+- `CC_UID` (default: *950*): id of the *codechecker* user which will be created
+during the image build and which will be used to start CodeChecker server.
+- `CC_GID` (default: *950*): id of the *codechecker* group which will be
+created during the image build.
+- `INSTALL_AUTH` (default: *no*): set it to `yes` to install authentication
+dependencies like *ldap*.
+- `INSTALL_PG8000` (default: *no*): set it to `yes` to install *pg8000*
+requirements.
+- `INSTALL_PSYCOPG2` (default: *no*): set it to `yes` to install *psycopg2*
+requirements.
+
+Example:
+```bash
+docker build \
+  --build-arg INSTALL_AUTH=yes \
+  --build-arg INSTALL_PSYCOPG2=yes \
+  --tag codechecker-server:latest .
+```
+
+## Usage
+To run a CodeChecker server docker container use the following command:
+```sh
+docker run -d \
+  -p 80:80 \
+  -v /home/username/codechecker_workspace:/workspace \
+  codechecker-server:latest
+```
+
+## Docker Compose
+[Docker Compose](https://docs.docker.com/compose/) is a tool for defining and
+running multi-container Docker applications.
+
+### `docker-compose.yml`
+CodeChecker server can be easily run by defining the services that make up your
+app in `docker-compose.yml` so they can be run together in an isolated
+environment.
+
+#### Sqlite setup
+```yaml
+version: '2'
+
+services:
+  codechecker:
+    image: codechecker-server:latest
+    ports:
+      - 8001:8001
+    volumes:
+      - /home/username/codechecker_workspace:/workspace
+```
+
+#### PostgreSQL setup
+```yaml
+version: '2'
+
+services:
+  db:
+    image: postgres:9.6-alpine
+    ports:
+      - '5433:5432'
+    volumes:
+      - /home/username/codechecker_database:/var/lib/postgresql/data
+    environment:
+      - POSTGRES_INITDB_ARGS=-E "SQL_ASCII"
```

Comment:
I not convinced that we should use `SQL_ASCII` here see postgreqsql encoding options: https://www.postgresql.org/docs/11/multibyte.html I would prefer if we would work with utf8.